### PR TITLE
Improve handling if user does not have access to the server

### LINKED ISSUE #3961: cs324 - users who haven't been added to the protected mode server get ugly exceptions
- User creates an account on cs324.codalab.org
- User gets ugly experience (can't create worksheets, etc.)

Ideal behavior:
- users should just be presented with a message that says "please contact the admins to get access to this server"

### PR DESCRIPTION
**Pull request by jzwang43**
### Reasons for making this change

    User creates an account on cs324.codalab.org
    User gets ugly experience (can't create worksheets, etc.)

### Related issues

Fixes #3961 

### Screenshots

Shows alert if user doesn't have access to the server:
image

### Checklist

* [x] I've added a screenshot of the changes, if this is a frontend change
* [x] I've added and/or updated tests, if this is a backend change
* [x] I've run the [pre-commit.sh](https://github.com/codalab/codalab-worksheets/blob/master/pre-commit.sh) script
* [x] I've updated [docs](https://github.com/codalab/codalab-worksheets/tree/master/docs), if needed

### CODE REVIEW

**Comment by epicfaace** [comment]
Let's show an error message like in https://worksheets.codalab.org/worksheets/0xbd86869844074a02aced04b5959153 for consistency:

image

**Comment by pranavjain** [comment]
@jzwang43 Are some unit tests still failing?

**Comment by jzwang43** [comment]
> @jzwang43 Are some unit tests still failing?

Seems to be a test issue not with the dockerfiles that I fixed.

**Comment by pranavjain** [comment]
Is this related to the test cases you and Ashwin wrote? What is the next step? @jzwang43 
Do let me know if you need some help in getting unblocked.

**Comment by jzwang43** [comment]
> Let's show an error message like in https://worksheets.codalab.org/worksheets/0xbd86869844074a02aced04b5959153 for consistency:
> 
> image

@epicfaace I have updated the code to use the error message you mentioned. Can I get a review on this? Thanks.

File: codalab/rest/schemas.py [hunk 1]
```diff
@@ -307,6 +306,7 @@ class UserSchema(Schema):
     url = fields.Url(allow_none=True)
     date_joined = fields.LocalDateTime("%c")
     avatar_id = fields.String(allow_none=True)
+    has_access = fields.String(allow_none=False)
```
Comment (epicfaace) [inline codalab/rest/schemas.py]: This should be a Boolean I think

**Review by epicfaace — CHANGES_REQUESTED** [review]

**Review by epicfaace — APPROVED** [review]

**Pull request merged**